include/fi_shm.h

```diff
@@ -108,15 +107,16 @@ struct smr_msg_hdr {
 
 union smr_cmd_data {
 	uint8_t			msg[SMR_MSG_DATA_LEN];
-	struct iovec		iov[SMR_MSG_DATA_LEN / sizeof(struct iovec)];
+	struct {
+		uint8_t		iov_count;
+		struct iovec	iov[(SMR_MSG_DATA_LEN - 8) /
+				    sizeof(struct iovec)];
+	} iov;
```

Comment:
Can drop the anonymous struct, especially since it results in typing iov.iov_count and iov.iov...